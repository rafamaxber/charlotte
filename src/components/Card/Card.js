import React from 'react';
import styled from 'styled-components';
import Button from '../../components/Button/Button';
import StarFullImage from '../../assets/star-filled.svg';

const Wrapper = styled.div`
  max-width: 940px;
  min-height: 318px;
  background: #FFFFFF;
  box-shadow: 0px 4px 17px rgba(0, 0, 0, 0.22);
  border-radius: 5px;
  margin: 10px auto;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
`;
const Image = styled.img`
  max-width: 247px;
  max-height: 247px;
`;
const WrapperDescription = styled.div`
  max-width: 402px;
  position: relative;
  &:after {
    content: "";
    width: 2px;
    height: 100%;
    right: -5%;
    top: 0;
    background-color: #D9D9D9;
    position: absolute;
    display: block;
  }
`;
const Rate = styled.div`
  margin-bottom: 10px;
`;
const Star = styled.img.attrs({
  src: StarFullImage,
})`
  width: 12.6px;
  margin-right: 4.3px;
`;
const Title = styled.div`
  font-family: 'Heebo', sans-serif; 
  font-style: normal;
  font-weight: bold;
  line-height: 26px;
  font-size: 16px;
  letter-spacing: 1.81818px;
  text-transform: uppercase;
  color: #F98100;
  margin-bottom: 20px;
`;
const Description = styled.div`
  font-family: 'Montserrat', sans-serif;
  font-style: normal;
  line-height: normal;
  font-size: 15px;
  color: #B5B5B5;
`;

const WrapperButtons = styled.div`
  margin-top: 30px;
`;
const WrapperPrice = styled.div`
  width: 160px;
  text-align: right;
  right: 3%;
  position: relative;
  margin-top: 5%;
`;
const PriceTitle = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: 300;
  line-height: normal;
  font-size: 16px;
  letter-spacing: 1.81818px;
  color: #B2B0C0;
`;
const Price = styled.div`
  font-family: 'Montserrat', sans-serif;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  font-size: 30px;
  letter-spacing: 3.40909px;
  color: #79BD1A;
`;

const CardButton = styled.div`
  margin-right: 20px;
  display: inline-block;
`;

export default ({ image, title, description, price, price_history, rate }) => (
  <Wrapper>
    <Image src={image}></Image>
    <WrapperDescription>
      <Rate>
      {
        Array.from(new Array(rate)).map((item, index) => <Star key={index}/>)
      }
      </Rate>
      <Title>{title}</Title>
      <Description>{description}</Description>
      <WrapperButtons>
        <CardButton>
          <Button size='small'>
            Book now
          </Button>
        </CardButton>
        <CardButton>
          <Button size='small' typeBtn="info">
            Price history
          </Button>
        </CardButton>
      </WrapperButtons>
    </WrapperDescription>
    <WrapperPrice>
      <PriceTitle>
        Total
      </PriceTitle>
      <Price>
        ${price}
      </Price>
    </WrapperPrice>
  </Wrapper>
);
