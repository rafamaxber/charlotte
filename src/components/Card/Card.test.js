import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const mock = [{ "name": "Facilities of SpringHill Suites by Marriott Charlotte Uptown", "description": "Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.", "image": "https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg", "rate": 5, "price": 400, "price_history": [{ "month": "Jan", "value": 438 }, { "month": "Feb", "value": 400.1 }, { "month": "Mar", "value": 425 }, { "month": "Apr", "value": 433.6 }, { "month": "May", "value": 428.4 }, { "month": "Jun", "value": 415 }, { "month": "Jul", "value": 412 }] }];
describe('Card', () => {
  it('Should match with snapshot', () => {
    const component = shallow(
      <Card
        image={mock[0].image}
        title={mock[0].name}
        description={mock[0].description}
        price={mock[0].price}
        price_history={mock[0].price_history}
        rate={mock[0].rate}
      ></Card>
    );
    expect(component).toMatchSnapshot();
  });
});
