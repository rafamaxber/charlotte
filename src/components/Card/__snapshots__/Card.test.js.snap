// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match with snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <LazyLoad
      height={200}
      offset={0}
      once={true}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <styled.img
        src="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
      />
    </LazyLoad>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.img />
          <styled.img />
          <styled.img />
          <styled.img />
          <styled.img />
        </styled.div>
        <styled.div>
          Facilities of SpringHill Suites by Marriott Charlotte Uptown
        </styled.div>
        <styled.div>
          Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.
        </styled.div>
        <styled.div>
          <styled.div>
            <Unknown
              size="small"
            >
              Book now
            </Unknown>
          </styled.div>
          <styled.div>
            <Unknown
              action={[Function]}
              size="small"
              typeBtn="info"
            >
              Price history
            </Unknown>
          </styled.div>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          Total
        </styled.div>
        <styled.div>
          $
          400
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>,
  "nodes": Array [
    <styled.div>
      <LazyLoad
        height={200}
        offset={0}
        once={true}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          src="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
        />
      </LazyLoad>
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.img />
            <styled.img />
            <styled.img />
            <styled.img />
            <styled.img />
          </styled.div>
          <styled.div>
            Facilities of SpringHill Suites by Marriott Charlotte Uptown
          </styled.div>
          <styled.div>
            Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.
          </styled.div>
          <styled.div>
            <styled.div>
              <Unknown
                size="small"
              >
                Book now
              </Unknown>
            </styled.div>
            <styled.div>
              <Unknown
                action={[Function]}
                size="small"
                typeBtn="info"
              >
                Price history
              </Unknown>
            </styled.div>
          </styled.div>
        </styled.div>
        <styled.div>
          <styled.div>
            Total
          </styled.div>
          <styled.div>
            $
            400
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Card
        description="Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking."
        image="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
        price={400}
        price_history={
          Array [
            Object {
              "month": "Jan",
              "value": 438,
            },
            Object {
              "month": "Feb",
              "value": 400.1,
            },
            Object {
              "month": "Mar",
              "value": 425,
            },
            Object {
              "month": "Apr",
              "value": 433.6,
            },
            Object {
              "month": "May",
              "value": 428.4,
            },
            Object {
              "month": "Jun",
              "value": 415,
            },
            Object {
              "month": "Jul",
              "value": 412,
            },
          ]
        }
        rate={5}
        title="Facilities of SpringHill Suites by Marriott Charlotte Uptown"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Card {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "description": "Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.",
          "image": "https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg",
          "price": 400,
          "price_history": Array [
            Object {
              "month": "Jan",
              "value": 438,
            },
            Object {
              "month": "Feb",
              "value": 400.1,
            },
            Object {
              "month": "Mar",
              "value": 425,
            },
            Object {
              "month": "Apr",
              "value": 433.6,
            },
            Object {
              "month": "May",
              "value": 428.4,
            },
            Object {
              "month": "Jun",
              "value": 415,
            },
            Object {
              "month": "Jul",
              "value": 412,
            },
          ],
          "rate": 5,
          "title": "Facilities of SpringHill Suites by Marriott Charlotte Uptown",
        },
        "refs": Object {},
        "state": Object {
          "showChart": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <LazyLoad
            height={200}
            offset={0}
            once={true}
            overflow={false}
            resize={false}
            scroll={true}
            unmountIfInvisible={false}
          >
            <styled.img
              src="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
            />
          </LazyLoad>
          <styled.div>
            <styled.div>
              <styled.div>
                <styled.img />
                <styled.img />
                <styled.img />
                <styled.img />
                <styled.img />
              </styled.div>
              <styled.div>
                Facilities of SpringHill Suites by Marriott Charlotte Uptown
              </styled.div>
              <styled.div>
                Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.
              </styled.div>
              <styled.div>
                <styled.div>
                  <Unknown
                    size="small"
                  >
                    Book now
                  </Unknown>
                </styled.div>
                <styled.div>
                  <Unknown
                    action={[Function]}
                    size="small"
                    typeBtn="info"
                  >
                    Price history
                  </Unknown>
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div>
                Total
              </styled.div>
              <styled.div>
                $
                400
              </styled.div>
            </styled.div>
          </styled.div>
        </styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <LazyLoad
            height={200}
            offset={0}
            once={true}
            overflow={false}
            resize={false}
            scroll={true}
            unmountIfInvisible={false}
          >
            <styled.img
              src="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
            />
          </LazyLoad>
          <styled.div>
            <styled.div>
              <styled.div>
                <styled.img />
                <styled.img />
                <styled.img />
                <styled.img />
                <styled.img />
              </styled.div>
              <styled.div>
                Facilities of SpringHill Suites by Marriott Charlotte Uptown
              </styled.div>
              <styled.div>
                Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking.
              </styled.div>
              <styled.div>
                <styled.div>
                  <Unknown
                    size="small"
                  >
                    Book now
                  </Unknown>
                </styled.div>
                <styled.div>
                  <Unknown
                    action={[Function]}
                    size="small"
                    typeBtn="info"
                  >
                    Price history
                  </Unknown>
                </styled.div>
              </styled.div>
            </styled.div>
            <styled.div>
              <styled.div>
                Total
              </styled.div>
              <styled.div>
                $
                400
              </styled.div>
            </styled.div>
          </styled.div>
        </styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Card
    description="Located 201 m from Mint Museum of Craft Design in Charlotte, SpringHill Suites by Marriott Charlotte Uptown features free WiFi access and private parking."
    image="https://www.raphaelfabeni.com.br/rv/test-resources/hotels/spring-hill-marriot.jpg"
    price={400}
    price_history={
      Array [
        Object {
          "month": "Jan",
          "value": 438,
        },
        Object {
          "month": "Feb",
          "value": 400.1,
        },
        Object {
          "month": "Mar",
          "value": 425,
        },
        Object {
          "month": "Apr",
          "value": 433.6,
        },
        Object {
          "month": "May",
          "value": 428.4,
        },
        Object {
          "month": "Jun",
          "value": 415,
        },
        Object {
          "month": "Jul",
          "value": 412,
        },
      ]
    }
    rate={5}
    title="Facilities of SpringHill Suites by Marriott Charlotte Uptown"
  />,
}
`;
